src/cli/commands/lint.ts

```diff
@@ -100,26 +109,38 @@ linting ./openapi.yaml
   }
 }
 
-async function lint(name: string, flags: any, command: Lint, customRules?: RuleCollection) {
+async function lint(name: string, flags: any, command: Lint, rules?: RuleCollection) {
   command.log(`Linting ${name}`);
   const spec: IParserResult = await readParsable(name, flags.encoding);
 
   const spectral = new Spectral();
-  if (customRules) {
-    command.log('Applying custom rules. Automatic rule detection is off.');
-    spectral.addRules(customRules);
-  } else if (parseInt(spec.data.swagger) === 2) {
-    command.log('OpenAPI 2.0 (Swagger) detected');
-    spectral.addFunctions(oas2Functions());
-    spectral.addRules(oas2Rules());
-  } else if (parseInt(spec.data.openapi) === 3) {
-    command.log('OpenAPI 3.x detected');
-    spectral.addFunctions(oas3Functions());
-    spectral.addRules(oas3Rules());
+  if (rules !== undefined) {
+    const numRules = new Array(rules).length;
```

Comment:
Does our collection have a `length` or `size` property or do we need to instantiate an array?

Reply:
I was getting errors when I used length.

Reply:
As I can see that Dictionary is just an object without any length or size properties. 
If so then wouldn't creating an Array from it always generate one element array regardless of how many properties the object has? Or do I see it wrong?

Reply:
> If so then wouldn't creating an array from it with new Array() always generate one element array regardless of how many properties the object has? Or do I see it wrong?

Yeah, it would.
```suggestion
    const numRules = Object.keys(rules).length;
```